ksql-api/src/main/java/io/confluent/ksql/api/plugin/PullQueryPublisher.java

```diff
@@ -19,24 +19,29 @@
 import io.confluent.ksql.api.server.BufferedPublisher;
 import io.confluent.ksql.api.spi.QueryPublisher;
 import io.vertx.core.Context;
+import java.util.ArrayList;
 import java.util.List;
 
 public class PullQueryPublisher extends BufferedPublisher<GenericRow> implements QueryPublisher {
 
   private final List<String> columnNames;
   private final List<String> columnTypes;
 
-  @SuppressWarnings({"unchecked", "rawtypes"})
   public PullQueryPublisher(final Context ctx, final TableRows tableRows,
       final List<String> columnNames, final List<String> columnTypes) {
-    super(ctx);
+    super(ctx, toGenericRows(tableRows));
+    this.columnNames = columnNames;
+    this.columnTypes = columnTypes;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Also, take defensive copies of the data, i.e. `ImmutableList.copyOf`

Reply:
ack